src/main/java/ui/components/pickers/LabelPickerState.java

```diff
@@ -0,0 +1,261 @@
+package ui.components.pickers;
+
+import backend.resource.TurboLabel;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * This class is used to represent the state of the label picker. In addition to representing a state,
+ * it contains the logic that handles every state transition
+ */
+public class LabelPickerState {
+
+    private Set<String> initialLabels;
+    private List<String> addedLabels;
+    private List<String> removedLabels;
+    private List<String> matchedLabels;
+    private List<TurboLabel> allLabels;
+    private OptionalInt currentSuggestionIndex;
+
+    @SuppressWarnings("PMD")
```

Comment:
Try `// NOPMD` on the method signature line? See if that works.

If not, I don't think there's a need for a specific format for documenting ignored warnings. Include it in the javadoc comment I guess.